### PR TITLE
Change pre-commit type definitions

### PR DESCRIPTION
**Pull request by adamchainz**
Django projects don't necessarily use a `templates/` folder to store their template files. I think it's better to use pre-commit's file identification feature to select all HTML files.  This comes from `identify`'s [extensions file](https://github.com/pre-commit/identify/blob/main/identify/extensions.py)).

Similarly, there are identify patterns for jinja (covering several extensions: `.j2`, `.jinja`, `.jinja2`), nunjucks, handlebars, and go templates (currently only `.gotmpl` rather than `.tmpl` - should `.tmpl` be added there?).

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
   entry: djlint --profile=handlebars
-  files: "\\.hbs"
+  types_or: [handlebars]
   language: python
 
 - id: djlint-golang
   name: djLint for Golang templates
   entry: djlint --profile=golang
-  files: "\\.tmpl"
+  types_or: [gotmpl]
```
Comment (adamchainz) [inline .pre-commit-hooks.yaml:28]: this change will currently stop the hook running on some go templates for users who upgrade - I would suggest `.tmpl` upstream in `identify` but I'm not sure if it's a "standard" - do you know?

**Comment by christopherpickering** [comment]
Thanks for the PR! This was originally added by [Rjevski](https://github.com/Rjevski).

@Rjevski  Do you have any comments?

Thanks!

**Comment by Rjevski** [comment]
With regards to the "identify" feature I think it's a good change. With regards to just including all HTML files it's a bit of a mixed bag because theoretically this will now apply to *all* HTML in a given repo, including non-templates. At the same time, I do agree that Django templates don't always live in `templates/`.

I wonder if there's a way to make this hook require explicit configuration (to force people to provide a location for their templates) so they can't just accidentally add it and wreak havoc across all their non-template files? If so I think it might be worth using that on all "ambiguous" file types where it's unclear whether it's a template.

**Comment by adamchainz** [comment]
It is possible for the hook to match no files, and force users to configure it. But I would assume that users are smart, and provide a wider default configuration. Users install the hook and run it and check the changes - they're responsible for ensuring it runs against the correct set of files.

**Pull request merged**